doc/development/deprecations.rst

```diff
@@ -9,6 +9,12 @@ deprecations are listed below.
 Pending deprecations
 --------------------
 
+* The ``QNode.get_best_method``, ``QNode.best_method_str`` and ``QNode.get_gradient_fn`` methods have been deprecated. 
+  Instead, use the ``qml.workflow.get_best_diff_method`` and ``qml.workflow._get_gradient_fn`` functions. 
```

Comment:
```suggestion
  Instead, use the ``qml.workflow.get_best_diff_method``. 
```

Given we will can change the call signature or delete `_get_gradient_fn` with no notice, we shouldn't include it here.  I think `get_best_diff_method` would be sufficient for any user purposes.

If we end up locking down `resolve_execution_config` this cycle, we can update this deprecation to also point to that function, but I think for now `get_best_diff_method` is sufficient for public consumption.

Reply:
Good point!